.github/workflows/build.yml

```diff
@@ -0,0 +1,53 @@
+name: Build
+on:
+  pull_request:
```

Comment:
Do we want this to run on every PR? The workflow takes about 15-20 minutes to complete.

https://github.com/bitcoindevkit/bdk-ffi/blob/master/.github/workflows/publish-android.yaml is a decent example of the full build + publish workflow that we'd want to turn this into eventually, which wouldn't run on every PR.